chain/chain.go

```diff
@@ -248,26 +248,24 @@ func (e *EthChainContext) Start() {
 
 		var client *ethclient.Client
 		var err error
-		for i := 0; i < types.EthereumDialMaxRetry; i++ {
+		for {
 			client, err = ethclient.Dial(e.DefaultConfig.W3external)
 			if err != nil || client == nil {
-				log.Warnf("cannot create a client connection: (%s), trying again (%d of %d)", err, i+1, types.EthereumDialMaxRetry)
-				time.Sleep(time.Second * 2)
+				log.Warnf("cannot create a client connection: (%s), trying again ...", err)
+				time.Sleep(time.Second * 5)
 				continue
 			}
 			break
 		}
-		if err != nil || client == nil {
-			log.Fatalf("cannot create a client connection: (%s), tried %d times.", err, types.EthereumDialMaxRetry)
-		}
+
 		tctx, cancel := context.WithTimeout(context.Background(), types.EthereumReadTimeout)
 		defer cancel()
 		nid, err := client.NetworkID(tctx)
 		if err != nil || nid == nil {
-			log.Fatalf("cannot get network ID from external web3: (%s)", err)
+			log.Errorf("cannot get network ID from external web3: (%s)", err)
```

Comment:
log.Error() does not stop the execution. It looks like this is not taken into account on this PR. Make the proper flow modifications to handle Errors instead of Fatals correctly.

Reply:
this is the case where the node is not synced  and we try to restart it but fails on close or fails on restart. Do we want to try again the restart or the close ? Or do we want to stop all the eth node operations without trying again these operations?